server/rules-engine/factHandler.js

```diff
@@ -0,0 +1,153 @@
+const jp = require('jsonpath')
+
+const sumallRuleHandler = function (event, almanac) {
+  almanac.factValue(event.params.calculation.sourcefact).then(source => {
+    let totalSum = 0
+    const values = jp.query(source, event.params.calculation.path)
+    totalSum = values.reduce((acc, value) => {
+      acc += value
+      return acc
+    }, totalSum)
+    almanac.addRuntimeFact(event.params.id, totalSum)
+  })
```

Comment:
All the example code for the library uses .then instead of await, so for simplicity I'd started out by using the same style.
I'm not familiar enough with Promises and async/await in node to avoid side effects so I kept with working code. It can be refactored later.

Reply:
Dealt with now, simpler change than I thought it would be. Code is clearer now.